app/src/store/data.ts

```diff
@@ -162,28 +165,46 @@ export default function useDataStore() {
     grants.value = grantsList as Grant[];
     grantRounds.value = grantRoundsList as GrantRound[];
 
-    const metaPtrs = grants.value.map((grant) => grant.metaPtr);
-    // Initialize metadata for metaPtrs we haven't encountered yet to status "pending"
+    // Fetch Grants Metadata
+    const grantMetaPtrs = grants.value.map((grant) => grant.metaPtr);
+    grantMetadata = await fetchMetaPtrs(grantMetaPtrs, grantMetadata);
+
+    // Fetch GrantRounds Metadata
+    const grantRoundMetaPtrs = grantRounds.value.map((grantRound) => grantRound.metaPtr);
+    grantRoundMetadata = await fetchMetaPtrs(grantRoundMetaPtrs, grantRoundMetadata);
```

Comment:
Ahh, I see why you used `const` instead of `let`. I think it's a misunderstanding of how refs work. Check out the [docs](https://v3.vuejs.org/api/refs-api.html), but basically when you defined a ref, you always make changes to it's `.value` property, not the object itself. 

The current approach is overriding the full ref, and will break the reactivity. Fixing that would give you code something like this:

```suggestion
    // Fetch Grants Metadata
    const grantMetaPtrs = grants.value.map((grant) => grant.metaPtr);
    grantMetadata.value = await fetchMetaPtrs(grantMetaPtrs, grantMetadata);

    // Fetch GrantRounds Metadata
    const grantRoundMetaPtrs = grantRounds.value.map((grantRound) => grantRound.metaPtr);
    grantRoundMetadata.value = await fetchMetaPtrs(grantRoundMetaPtrs, grantRoundMetadata);
```

But, that's not exactly what we want either, as partly explained [here](https://github.com/dcgtc/dgrants/pull/86/#discussion_r691600963). The current approach has two issues:
1. `grantRoundMetadata` is blocked on waiting for `grantMetadata` to resolve
2. Individual metadata pointers may resolve slowly, and the current approach blocks the UI until all metadata is resolved

This is an easy fix for you though, basically just change the snippet to

```suggestion
    // Fetch metadata
    const grantMetaPtrs = grants.value.map((grant) => grant.metaPtr);
    const grantRoundMetaPtrs = grantRounds.value.map((grantRound) => grantRound.metaPtr);
    void fetchMetaPtrs(grantMetaPtrs, grantMetadata);
    void fetchMetaPtrs(grantRoundMetaPtrs, grantRoundMetadata);
```

Reply:
Also maybe just add a few more comments around the `fetchMetaPtrs`, because I was confused at first since it was mutating state _and_ returning something. But we only need it to mutate state, and the state being mutated is the ref passed as the `metadata` input. Worth detailing that + specifying the second input should be a `ref`

Reply:
mind need a hand in figuring out how to word it 😅

Reply:
Sample below:

```typescript
/**
 * @notice Helper method that fetches metadata for a Grant or GrantRound, and saves the data
 * to the state as soon as it's received
 * @param metaPtrs Array of URLs to resolve
 * @param metadata Name of the store's ref to assign resolve metadata to
 */
async function fetchMetaPtrs(metaPtrs: string[], metadata: Ref) {
  const newMetadata = metaPtrs
    .filter((metaPtr) => !metadata.value[metaPtr])
    .reduce((prev, cur) => {
      return { ...prev, [cur]: { status: 'pending' } };
    }, {});
  // save these pending metadata objects to state
  metadata.value = { ...metadata.value, ...newMetadata };
  // resolve metadata via metaPtr and update state
  Object.keys(newMetadata).map(async (url) => {
    try {
      const data = await resolveMetaPtr(url);
      metadata.value[url] = { status: 'resolved', ...data };
    } catch (e) {
      metadata.value[url] = { status: 'error' };
      console.error(e);
    }
  });
}
```